src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtils.java

```diff
@@ -0,0 +1,176 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.utils;
+
+import java.io.IOException;
+import java.lang.reflect.Modifier;
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.common.reflect.ClassPath;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck;
+import com.puppycrawl.tools.checkstyle.api.AutomaticBean;
+import com.puppycrawl.tools.checkstyle.api.BeforeExecutionFileFilter;
+import com.puppycrawl.tools.checkstyle.api.Filter;
+import com.puppycrawl.tools.checkstyle.api.RootModule;
+
+/**
+ * Contains utility methods for module reflection.
+ * @author LuoLiangchen
+ */
+public final class ModuleReflectionUtils {
+
+    /** Prevent instantiation. */
+    private ModuleReflectionUtils() {
+    }
+
+    /**
+     * Gets checkstyle's non-abstract checks in the given package recursively.
+     * @param packageName the package name to use
+     * @param loader the class loader used to load Checkstyle package name
+     * @return the set of checkstyle's check classes
+     * @throws IOException if the attempt to read class path resources failed
+     * @see #isValidCheckstyleClass(Class)
+     * @see #isCheckstyleCheck(Class)
+     */
+    public static Set<Class<?>> getCheckstyleChecksRecursive(
```

Comment:
@romani `getCheckstyleChecksRecursive` served our needs in test.
My question is do we need this method in production. Currently no other production code calls this method so we are including it only for 3rd party use. It does the same thing as `getCheckstyleModulesRecursive` but restricts results to only checks.